python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,921 @@
+</div>
+
+### Quiz: Lists
+
+1. While of the following is FALSE about lists? Select all that apply.
+
+    [( )] Lists are changeable; items can be added, removed, or replaced after the list is created.
+    [( )] Lists can be assigned to variables.
+    [( )] Lists can contain a mix of data types.
+    [(X)] Lists are unordered; you cannot access an item in the list by its position.
+    ***
+    <div class = "answer">
+    Lists are chanageable, can be stored in varibles for later use, can contain a mix of data types, and they are **ordered**, meaning that items remain in the position that they were put into the list, and you can access or edit list items by value **or** position (also called the **index**).
+    </div>
+    ***
+
+2. Given the following list of numbers, what is the **index** of the number **15**?
+
+    `[2, 14, 9, 101, 15, 37]`
+
+    [( )] 5
+    [(X)] 4
+    [( )] 15
+    [( )] 6
+    ***
+    <div class = "answer">
+    Indexing in Python begins with **0**, not 1, and so the index of 15, which is the fifth item in the list, is **4**.  
+    </div>
+    ***
+
+## Loops
+
+A Python list is an **iterable** object, or an object whose members can be returned one at a time. This is important in **loops**, which repeat (or iterate) the same operation for each element in an iterable object like a list.
+
+For a sense of how loops work in Python, here is a simple example. Let's say there are five children at a party, and they each start with a certain number of pieces of candy. Next, let's say that we give each child 5 more pieces of candy. Using a simple loop, we can get a list of how many total pieces of candy each child has now using the `for` and `in` keywords.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+starting_candy = [3, 10, 11, 6, 7]
+candy_day1 = [] #Here we are initiating an empty list, so that we can add elements to it later
+for i in starting_candy: #In the loop, i will take on the value of each list element in turn
+  j = i + 5
+  candy_day1.append(j)
+print(candy_day1)
+
+</script>
+</lia-keep>
+</div>
+
+<div class = "important">
+Did you notice the explanatory text in the above code cell that started with a pound sign? The pound sign ( # ) in a code cell indicates the start of a **comment**. Comments aren't recognized as code and won't be run. Each line of a comment must be marked with a new pound sign. Comments are an excellent way to add brief explanations and clarifications about your code.
+</div>
+
+This example loops through the list of numbers, adds 5 to each number one at a time, and adds the sum to a new list using the append() method for lists. Finally, we printed the new list to our screen. This kind of loop is sometimes called a for loop; there is another kind of loop called a while loop, which is often used when we don't know the number of times we'll have to iterate through a block of code before we start (check out this page for more information about while loops).
```

Comment:
```suggestion
This example loops through the list of numbers, adds 5 to each number one at a time, and adds the sum to a new list using the append() method for lists. Finally, we printed the new list to our screen. This kind of loop is sometimes called a **for loop**; there is another kind of loop called a **while loop**, which is often used when we don't know the number of times we'll have to iterate through a block of code before we start (check out this page for more information about while loops).
```

Also should there be a link to learn more about while loops?

Reply:
Oh yeah probably!